packages/amplify_auth_cognito/android/src/main/kotlin/com/amazonaws/amplify/amplify_auth_cognito/device/DeviceHandler.kt

```diff
@@ -0,0 +1,102 @@
+package com.amazonaws.amplify.amplify_auth_cognito.device
+
+import com.amazonaws.amplify.amplify_auth_cognito.AuthErrorHandler
+import com.amazonaws.amplify.amplify_auth_cognito.base.AtomicResult
+import com.amazonaws.mobile.client.AWSMobileClient
+import com.amazonaws.mobile.client.Callback
+import com.amazonaws.mobile.client.results.ListDevicesResult
+import com.amplifyframework.auth.AuthDevice
+import com.amplifyframework.auth.AuthException
+import com.amplifyframework.auth.cognito.util.CognitoAuthExceptionConverter
+import com.amplifyframework.core.Amplify
+import io.flutter.plugin.common.MethodCall
+import io.flutter.plugin.common.MethodChannel
+import kotlinx.coroutines.*
+
+/**
+ * Handles method calls for the Devices API.
+ */
+class DeviceHandler(private val errorHandler: AuthErrorHandler) :
+    MethodChannel.MethodCallHandler {
+    companion object {
+        /**
+         * Methods this handler supports.
+         */
+        private val methods = listOf("rememberDevice", "forgetDevice", "fetchDevices")
+
+        /**
+         * Whether this class can handle [method].
+         */
+        fun canHandle(method: String): Boolean = methods.contains(method)
+    }
+
+    /**
+     * Scope for running asynchronous tasks.
+     */
+    private val scope = CoroutineScope(Dispatchers.IO) + CoroutineName("DeviceHandler")
```

Comment:
Coroutines became stable in Kotlin version 1.3. Do we know if it's possible whether any existing customers may be using an older version of Kotlin? Would our library otherwise work if they were using an older version?

Reply:
I believe that apps and plugins can use different versions of Kotlin. But this would only be a potential issue for projects generated before 1/24/19, according to the [git history](https://github.com/flutter/flutter/commits/stable/packages/flutter_tools/templates/app/android-kotlin.tmpl/build.gradle).